src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -1234,13 +1235,14 @@ internal bool Compile(bool optimized)
 
         internal static void LogScriptBlockCreation(ScriptBlock scriptBlock, bool force)
         {
-            if (force || ShouldLogScriptBlockActivity("EnableScriptBlockLogging"))
+            ScriptBlockLogging logSetting = GetScriptBlockLoggingSetting();
+            if (force || logSetting?.EnableScriptBlockLogging == true)
             {
                 if (!scriptBlock.HasLogged || InternalTestHooks.ForceScriptBlockLogging)
                 {
                     // If script block logging is explicitly disabled, or it's from a trusted
                     // file or internal, skip logging.
-                    if (ScriptBlockLoggingExplicitlyDisabled() ||
+                    if (logSetting?.EnableScriptBlockLogging == false ||
                         scriptBlock.ScriptBlockData.IsProductCode)
```

Comment:
Theoretically, this can be re-written as
```csharp
if ((force && logSetting?.EnableScriptBlockLogging != false) || logSetting?.EnableScriptBlockLogging == true)
{
    if ((!scriptBlock.HasLogged || InternalTestHooks.ForceScriptBlockLogging) && !scriptBlock.ScriptBlockData.IsProductCode)
    {
        ....
    }
}
```
Or even more aggressively, these two `if` blocks can be merged into one. But this doesn't look very readable, so I think I'm fine with the original code.